opal/corelib/file.rb

```diff
@@ -30,6 +30,18 @@ def basename(path)
       split(path)[-1]
     end
 
+    def extname(path)
+      raise TypeError, 'no implicit conversion of nil into String' if path.nil?
+      path = path.to_path if path.respond_to?(:to_path)
+      raise TypeError, "no implicit conversion of #{path.class} into String" unless path.is_a?(String)
+      filename = basename(path)
```

Comment:
`File.basename` method is not fully compliant with the Ruby specs but all tests passed nevertheless :wink: